src/domain/usecases/GetDiseaseOutbreakWithOptionsUseCase.ts

```diff
@@ -0,0 +1,72 @@
+import { FutureData } from "../../data/api-futures";
+import { DiseaseOutbreakEventBaseAttrs } from "../entities/disease-outbreak-event/DiseaseOutbreakEvent";
+import { DiseaseOutbreakEventWithOptions } from "../entities/disease-outbreak-event/DiseaseOutbreakEventWithOptions";
+import { Future } from "../entities/generic/Future";
+import { Id } from "../entities/Ref";
+import { DiseaseOutbreakEventRepository } from "../repositories/DiseaseOutbreakEventRepository";
+import { OptionsRepository } from "../repositories/OptionsRepository";
+import { TeamMemberRepository } from "../repositories/TeamMemberRepository";
+
+export class GetDiseaseOutbreakWithOptionsUseCase {
+    constructor(
+        private options: {
+            diseaseOutbreakEventRepository: DiseaseOutbreakEventRepository;
+            optionsRepository: OptionsRepository;
+            teamMemberRepository: TeamMemberRepository;
+        }
+    ) {}
+
+    public execute(id?: Id): FutureData<DiseaseOutbreakEventWithOptions> {
+        if (id) {
+            return this.options.diseaseOutbreakEventRepository
+                .get(id)
+                .flatMap(diseaseOutbreakEventBase => {
+                    return this.getDiseaseOutbreakEventWithOptions(diseaseOutbreakEventBase);
+                });
+        } else {
+            return this.getDiseaseOutbreakEventWithOptions();
+        }
+    }
+
+    private getDiseaseOutbreakEventWithOptions(
+        diseaseOutbreakEventBase?: DiseaseOutbreakEventBaseAttrs
+    ): FutureData<DiseaseOutbreakEventWithOptions> {
+        return Future.joinObj({
+            hazardTypes: this.options.optionsRepository.getAllHazardTypes(),
+            mainSyndromes: this.options.optionsRepository.getAllMainSyndromes(),
+            suspectedDiseases: this.options.optionsRepository.getAllSuspectedDiseases(),
+            notificationSources: this.options.optionsRepository.getAllNotificationSources(),
+            incidentStatus: this.options.optionsRepository.getAllIncidentStatus(),
+            teamMembers: this.options.teamMemberRepository.getAll(),
+        }).flatMap(
```

Comment:
@tokland we have just had the meeting with jorge about this and to fix this it would be necessary to change the following:
- Remove optionsRepository and every options will be an entity: DataSource, HazardType, MainSyndrome, SuspectedDisease, NotificationSource and IncidentStatus. Each one will have its own repository and use case and these use cases will be called in the hook that will be in charge of creating the necessary options for being used in the form.
- Delete `GetDiseaseOutbreakWithOptionsUseCase`  and create one entity for the ErrorMessages and another for the Rules (both will come from Datastore), with its own repository and use case, and these use cases will be called in the form that needs it
- Now we just have the presentation validation of the form rule with the data sources, so we need to add also an entity validation for rule: when data source IBS then HazardType empty and MainSyndrome and SuspectedDisease with value , if data source EBS then MainSyndrome and SuspectedDisease empty and HazardType with value

As there are more PRs that depend on this PR and https://github.com/EyeSeeTea/zebra-dev/pull/9, we have decided that these changes will go in the following PR, in order to merge this one as soon as possible. (cc @9sneha-n )

Reply:
Ok!